storage/local/persistence.go

```diff
@@ -683,6 +691,11 @@ func (p *persistence) checkpointSeriesMapAndHeads(fingerprintToSeries *seriesMap
 
 	var realNumberOfSeries uint64
 	for m := range iter {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
```

Comment:
We might want a better error message here, as I can see this causing unnecessary concern on the part of users at shutdown.